implementations/rust/ockam/ockam_app_lib/src/incoming_services/state.rs

```diff
@@ -0,0 +1,365 @@
+use crate::state::{AppState, ModelState};
+use minicbor::{Decode, Encode};
+use ockam_api::cloud::share::{InvitationWithAccess, ServiceAccessDetails};
+use ockam_api::error::ApiError;
+use ockam_api::identity::EnrollmentTicket;
+use serde::{Deserialize, Serialize};
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use tracing::warn;
+
+#[derive(Clone, Debug, Decode, Encode, Serialize, Deserialize, PartialEq)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct PersistentIncomingServiceState {
```

Comment:
In terms of overall structure I wonder if we wouldn't be better off if we:
 
 - grab all the invitations from the controller
 - transform them to `IncomingService` with additional `custom_name / enabled` fields for every new invitation received
 - persist the `IncomingServices` directly
 - recreate/restart the inlets directly from the persisted data

This would get rid of one data structure but also give more reliability to the app which would also be able to work even if the controller has hiccups when retrieving invitations.

Reply:
I thought the same, but with the objective to lower the application bootstrap latency and avoid the "empty services" in the first second.

My arguments against it are:
The application only makes sense when it's connected to the orchestrator, if the controller is down or having issues, we want direct feedback, and by serializing more information, we increase the possibility to break back-compatibility. I don't think it's worth doing it during this initial phase.

Reply:
> The application only makes sense when it's connected to the orchestrator,

Let's bring that point up in our next meeting